caas/kubernetes/provider/credentials.go

```diff
@@ -40,7 +42,15 @@ func (environProviderCredentials) DetectCredentials() (*cloud.CloudCredential, e
 	if err != nil {
 		return nil, errors.Trace(err)
 	}
-	caasConfig, err := clientConfigFunc()
+
+	kubeConfigPath := clientconfig.GetKubeConfigPath()
+	f, err := os.Open(kubeConfigPath)
+	if err != nil {
+		return nil, errors.Trace(err)
```

Comment:
We need to ensure that an error satisfying errors.IsNotFound() is returned here if the file doesn't exist. That way the caller of DetectCredentials will not error out, it will know the creds don't exist